org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -0,0 +1,367 @@
+* - Peter Takacs, Cloudempiere                                        *
+**********************************************************************/
+package org.adempiere.webui.editor;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.ArrayList;
+
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.webui.ValuePreference;
+import org.adempiere.webui.apps.AEnv;
+import org.adempiere.webui.component.Textbox;
+import org.adempiere.webui.component.ToolBarButton;
+import org.adempiere.webui.event.ContextMenuEvent;
+import org.adempiere.webui.event.ContextMenuListener;
+import org.adempiere.webui.event.ValueChangeEvent;
+import org.adempiere.webui.theme.ThemeManager;
+import org.adempiere.webui.util.GridTabDataBinder;
+import org.adempiere.webui.window.WFieldRecordInfo;
+import org.adempiere.webui.window.WRecordIDDialog;
+import org.compiere.model.GridField;
+import org.compiere.model.MColumn;
+import org.compiere.model.MQuery;
+import org.compiere.model.MRole;
+import org.compiere.model.MSysConfig;
+import org.compiere.model.MTable;
+import org.compiere.model.PO;
+import org.compiere.model.Query;
+import org.compiere.util.DB;
+import org.compiere.util.Env;
+import org.compiere.util.Msg;
+import org.compiere.util.Util;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zk.ui.event.Events;
+import org.zkoss.zul.Div;
+import org.zkoss.zul.Vlayout;
+
+/**
+ * 
+ * @author Peter Takacs, Cloudempiere
+ *
+ */
+public class WRecordIDEditor extends WEditor implements ContextMenuListener, IZoomableEditor {
+	
+	private boolean m_ReadWrite;
+	private Object oldValue;
+	private Object oldTableID;
+	
+	private GridField tableIDGridField;
+	private WEditor tableIDEditor;
+	private GridTabDataBinder dataBinder;
+	
+	private Div wrapperDiv;
+	private Textbox recordTextBox;
+	private ToolBarButton editButton;
+	private ToolBarButton zoomButton;
+	
+	protected String selectedTableName = "";
+	protected String selectedRecordIdentifyer = "";
+	
+	private static final String IMAGES_CONTEXT_ZOOM_PNG = "images/Zoom16.png";
+	private static final String IMAGES_CONTEXT_EDIT_RECORD_PNG = "images/EditRecord16.png";
+
+	public WRecordIDEditor(GridField gridField, boolean tableEditor, IEditorConfiguration editorConfiguration) {
+		super(new Vlayout(), gridField, tableEditor, editorConfiguration);
+		init();
+	}
+
+	private void init() {
+		wrapperDiv = new Div();
+		wrapperDiv.setSclass("recordid-editor");
+		wrapperDiv.setParent(this.getComponent());
+		wrapperDiv.addEventListener(Events.ON_RIGHT_CLICK, this);
+
+		if(gridTab != null) {
+			tableIDGridField = gridTab.getField("AD_Table_ID");
+			tableIDEditor = WebEditorFactory.getEditor(tableIDGridField, true);
+			
+			if(tableIDGridField.getValue() != null)
+				tableIDEditor.setValue(tableIDGridField.getValue());
+						
+			dataBinder = new GridTabDataBinder(gridTab);
+			tableIDEditor.addValueChangeListener(dataBinder);
```

Comment:
why do we need to do this here ? this should the job of the field editor for AD_Table_ID field.